ProcessLib/ProcessVariable.h

```diff
@@ -56,10 +57,20 @@ class ProcessVariable
 	/// Returns a mesh on which the process variable is defined.
 	MeshLib::Mesh const& getMesh() const;
 
-	void initializeDirichletBCs(MeshGeoToolsLib::MeshNodeSearcher& searcher,
-	        const AssemblerLib::LocalToGlobalIndexMap& dof_table,
-	        const unsigned nodal_dof_idx,
-	        std::vector<GlobalIndexType>& global_ids, std::vector<double>& values);
+	template <typename OutputIterator>
+	void initializeDirichletBCs(
+	    OutputIterator output_bcs,
+	    MeshGeoToolsLib::MeshNodeSearcher& searcher,
+	    const AssemblerLib::LocalToGlobalIndexMap& dof_table,
+	    const unsigned nodal_dof_idx)
+	{
+		for (auto& bc_config : _dirichlet_bc_configs)
+		{
+			DirichletBc<GlobalIndexType> bc;
+			bc_config->initialize(searcher, dof_table, nodal_dof_idx, bc);
+			output_bcs = bc;  // output iterator
```

Comment:
Different question: Do you have to increment the output iterator? I'm not used to it, so I don't know.